### ISSUE TITLE
Contribution guidelines

### ISSUE THREAD

**Issue by AdrienLemaire** [awesome-global-contributions/awesome-global-contributions#2]
Related: #1

<details>
<summary>Examples</summary>

- [awesome/contributing](https://github.com/sindresorhus/awesome/blob/master/contributing.md)
- [awesome-android/contributing](https://github.com/JStumpp/awesome-android/blob/master/contributing.md)
- [awesome-nodejs/contributing](https://github.com/sindresorhus/awesome-nodejs/blob/master/contributing.md)
 - [frontend-dev-bookmarks/contributing](https://github.com/dypsilon/frontend-dev-bookmarks/blob/master/contributing.md)
- [awesome-javascript](https://github.com/sorrycc/awesome-javascript/blob/master/CONTRIBUTING.md)
- [awesomedata/apd-core](https://github.com/awesomedata/apd-core/wiki) !interesting

</details>

**To do:**
- [ ] Define contribution guidelines
- [ ] Define maintainers rules
- [ ] Write contribution documentation
- [ ] Write Issue and PR templates
- [ ] Define Github labels
- [ ] Merge to master and change repo status to [![Project Status: WIP – Initial development is in progress, but there has not yet been a stable, usable release suitable for the public.](https://www.repostatus.org/badges/latest/wip.svg)](https://www.repostatus.org/#wip)

**Contribution rules proposal:**
- [ ] Align to the upstream [contributing rules](https://github.com/sindresorhus/awesome/blob/master/contributing.md), [code of conduct](https://github.com/sindresorhus/awesome/blob/master/code-of-conduct.md) and [manifesto](https://github.com/sindresorhus/awesome/blob/master/awesome.md#pay-attention-to-grammar) (eg "pay attention to grammar", "ensure all entries are consistent")
- [ ] Familiarity with Github & markdown syntax needed
- [ ] Entry Format #3
- [ ] script-friendliness
- [ ] Respect of issue & PR templates
- [ ] Proper usage of Github labels

**Maintainer rules proposal:**
- [ ] GPG signed commits?
- [ ] Good understanding of Github, git, OSS & SDGs required
- [ ] Defined responsibilities
- [ ] Anyone can join, the more the merrier?

**Comment by AdrienLemaire** [comment]
It would be useful to write a script to regularly update the list:
- check for dead links
- update details for entries (activity, numbers)

If we agree to maintain the script as well, we then need to add details to the contribution guidelines for new entries to be script-friendly.

**Comment by OscarAblinger** [comment]
As you have mentioned (#3) there are multiple things that people might want to look at to search for projects that they want to contribute in:
- Programming Language(s) used (Note that I don't think that it will be possible
- SDGs
- Activity
- Beginner friendliness (Do they have a contribution guide? Good first issues?)
- etc.

This means, that a website to filter/search is rather important. (And it should be super easy to create. I can do that, once we've defined everything related to that)

For the list itself, I agree with the sorting by SDGs. It seems to be the most sensible.

**Because of this, I would suggest using a JSON file for all the contributions.**
The main point for this would be that the website could easily request the list and process it – through this it will always have the most up-to-date data without needing to change anything.
However, I also see other advantages:
1) It is automatically script-friendly (given a valid JSON of course)
1) We can generate the README: I think this is a slight plus since it should make it easier to adjust formatting as we go forward: We already have a lot of interesting points that we want to know about each project so it's not unlikely that we will have to adjust a bit depending on what data will generally be provided.
2) Users submissions are easier: Creating a JSON is really easy for everyone and we can easily lint that. Therefore we won't have to deal with formatting at all (except in multi-line descriptions). Additionally it would be possible to create that through a simple web form (is that useful?). We could even possibly let users submit entries over the website? I don't know how useful that would be, since _most_ programmers are able to use Github and I would guess that not many non-programmers will want to contribute to this list.
3) We can generate a CONTRIBUTERS file (instead of them adding (and thanking) themselves to it, which some might feel weird doing)

A reason against a JSON-file as the SSOT is the point that people won't have a good preview of the result, but I don't think that will be a big problem.

**Comment by OscarAblinger** [comment]
Referring to the rules proposal:
- A good step-by-step contribution guide is of course needed
- The Code of Conduct is definitely a good thing to include. The Manifesto is only really relevant for maintainers, right?
- Familiarity with Markdown, Entry Formatting and script-friendliness would mostly be given with a JSON file
- Respect of templates & Github labels also make complete sense

For the maintainer rules proposal:
- I'm generally not a fan of GPG and am not really sure what that would accomplish 🤔
- Good understanding makes sense
- I agree with the "the more the merrier" sentiment for now – if the list becomes very popular this might have to be revised, though. But in the beginning it's probably more important to get as many people on board as possible
- Concerning the responsibilities, I think we have the following:
  - Control & Merge PR of other contributors (**This includes checking the validity of the entry**)
  - Narrate the issues etc.
  - Create and maintain scripts, the website & the guidelines
  - Potentially spreading the word? This should be unintrusive things like reddit posts.
  - If the entry is valid, but lacks some secondary information, we could also try and reach the projects themselves to complete the entry?

  Did I forget any responsibilities or do you disagree with some?

**Comment by AdrienLemaire** [comment]
@AblingerOscar thanks for offering to build and maintain the related website, this sounds awesome!

Can I suggest that we use YAML? It's more human-friendly, allows for inline comments, etc, and perfect compatibility with JSON.
We can also have a list of YAML files that can then be concatenated together (I assume you know about jq and yq).

Auto-generating the README from YAML files LGTM!

> I'm generally not a fan of GPG

Ok, let's not mind about it then.

> I agree with the "the more the merrier" sentiment for now

Great! I have a friend @Keats who's interested in projects helping to resolve the global warming crisis, I'll try to invite him. But he's already involved in many FOSS projects, so he might be too busy to help.

**Comment by OscarAblinger** [comment]
> Can I suggest that we use YAML?

Sure, it also has better support for multi-line strings, so that sounds good to me.

> We can also have a list of YAML files that can then be concatenated together

If that's more maintainable. But what would those files be? One for every goal?

> I assume you know about jq and yq

Heard of jq, but never used it. I've never heard about yg (my contact with yaml was limited to a few configurations that I wrote, so I never had a need for it). It seems easy enough to use, though, thanks for mentioning.

> Great! I have a friend @Keats who's interested in projects helping to resolve the global warming crisis, I'll try to invite him.

Great 👍

**Comment by AdrienLemaire** [comment]
We can continue this discussion in Slack. Here's [the invite link](https://join.slack.com/t/awesomeglobal-7zk1378/shared_invite/enQtNzA3NTcyMTQwMzU1LTU4OGUyNzRhNjQyYjk5NTRkZDU1NjliZmRmMmY0MzIzM2IwYmRlNDM3Nzg1YjE1ZmRmNTA3NjBlMzI2YTA2MmY)

slack urls are limited in characters, so I went with `https://awesome-contributions.slack.com/`

**Comment by OscarAblinger** [comment]
I've pushed a first draft for a contribution guide on branch [feature/contribution-guideline](https://github.com/awesome-global-contributions/awesome-global-contributions/blob/feature/contribution-guideline/CONTRIBUTING.md).
It has a lot of TODOs based on other open issues, but the general idea is there.

It also has a updated yaml template  (#3) based on things I had to think about during creation of the contribution.
